app/src/main/java/com/github/studydistractor/sdp/createActivity/CreateActivityViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.studydistractor.sdp.createActivity
+
+import com.github.studydistractor.sdp.ui.state.CreateActivityUiState
+import com.google.android.gms.tasks.Task
+import kotlinx.coroutines.flow.StateFlow
+
+interface CreateActivityViewModel {
+
+    val uiState: StateFlow<CreateActivityUiState>
```

Comment:
Since you will always override it, it's probably not useful to have it here.

Reply:
I use it for the CreateActivityScreen, so it needs to be there. But if you have a suggestion on how to do otherwise, i'm open to it.

Reply:
If you make CreateActivityViewModel and CreateActivityUiState abstract classes, you can do this instead : https://stackoverflow.com/a/45756617 and have both event and distraction uiStates inherit from activityUiState. That way you can use the uiState in CreateActivityViewModel without risking EventUiState and DistractionUiState diverging from ActivityUiState

Reply:
i dun think so, this is all good :)